marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -204,6 +221,35 @@ private JsonNode determineReadRowsEndpointParams(Map<String, String> options) {
         return endpointParams;
     }
 
+    private void loadModuleIfNotPresent(String modulePath, Format format) {
```

Comment:
Just realized this is duplicated from HubDataSourceWriter!

Let's instead have a ModuleWriter class with methods of loadModuleIfNotPresent and endpointExists. It should go in the "v2" package since it's generic to read/write. Its constructor should accept a HubClient which it'll use to interact with the modules database.

Reply:
Done